src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -384,6 +377,19 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count)
 		(*destination->rStartup)(destination, operation, queryDesc->tupDesc);
 	}
 
+	/* we only support returning nothing or scanning forward */
```

Comment:
To avoid goto` ```
if ( ScanDirectionIsForward(direction) ) 
{
    /* lines 397-445 */
}
else if ( ! ScanDirectionIsNoMovement(direction) )
{
    /* lines 388-390 */
}

/* no-movement scan direction requires no action */

/* lines 449 and onward */
```